astropy/wcs/src/wcslib_prjprm_wrap.c

```diff
@@ -0,0 +1,1008 @@
+
+    if (!PyArg_ParseTupleAndKeywords(args, kwds, "OO:set_pvi",
+        (char **)keywords, &index, &value)) {
+        return NULL;
+    }
+
+    if (!PyLong_Check(index)) {
+        PyErr_SetString(PyExc_TypeError,
+        "PV index must be an integer number.");
+    }
+
+    idx = PyLong_AsLong(index);
+    if (idx == -1 && PyErr_Occurred()) {
+        return NULL;
+    }
+
+    if (idx < 0 || idx >= PVN) {
+        PyErr_Format(PyExc_ValueError,
+            "PV index must be an integer number between 0 and %d.", PVN - 1);
+        return NULL;
+    }
+
+    if (value == Py_None) {
+        /* If pv is set to None - reset pv to prjini values: */
+        self->x->pv[idx] = (idx > 0 && idx < 4) ? UNDEFINED : 0.0;
+        self->x->flag = 0;
+        if (self->owner) ((PyCelprm*)self->owner)->x->flag = 0;
+        Py_RETURN_NONE;
+    }
+
+    if (PyFloat_Check(value) || PyLong_Check(value)) {
+        data = PyFloat_AsDouble(value);
+        if (data == -1.0 && PyErr_Occurred()) {
+            return NULL;
+        }
+
+    } else if (PyUnicode_Check(value)) {
+        flt_value = PyFloat_FromString(value);
+        if (!flt_value) return NULL;
+        data = PyFloat_AsDouble(flt_value);
+        Py_DECREF(flt_value);
+        if (data == -1.0 && PyErr_Occurred()) {
+            return NULL;
+        }
+
+    } else {
+        if (PyArray_Converter(value, &value_array) == NPY_FAIL) {
+            return NULL;
+        }
+
+        size = PyArray_SIZE(value_array);
+        if (size != 1) {
+            Py_DECREF(value_array);
+            PyErr_SetString(PyExc_ValueError,
+                "PV value must be a scalar-like object or None.");
+            return NULL;
+        }
+
+        scalar = PyArray_ToScalar(PyArray_DATA(value_array), value_array);
+        Py_DECREF(value_array);
+        if (!scalar) {
+            Py_DECREF(scalar);
+            PyErr_SetString(PyExc_TypeError, "Unable to convert value to scalar.");
+        }
+
+        PyArray_CastScalarToCtype(scalar, &data, dbl_descr);
+        Py_DECREF(scalar);
+        if (PyErr_Occurred()) {
+            return NULL;
+        }
+    }
+
+    data = (isnan(data)) ? UNDEFINED : data;
+
+    if (!is_dbl_equal(self->x->pv[idx], data)) {
+        self->x->flag = 0;
+        if (self->owner) ((PyCelprm*)self->owner)->x->flag = 0;
+    }
+    self->x->pv[idx] = data;
+
+    Py_RETURN_NONE;
+}
+
+
+static PyObject* PyPrjprm_get_bounds(PyPrjprm* self, void* closure)
+{
+    if (is_prj_null(self)) {
+        return NULL;
+    } else {
+        return get_int("bounds", self->x->bounds);
+    }
+}
+
+
+static int PyPrjprm_set_bounds(PyPrjprm* self, PyObject* value, void* closure)
+{
+    if (is_prj_null(self) || is_readonly(self)) {
+        return -1;
+    } else if (value == Py_None) {
+        self->x->bounds = 0;
+        return 0;
+    } else {
+        return set_int("bounds", value, &self->x->bounds);
+    }
+}
+
+
+static PyObject* PyPrjprm_get_w(PyPrjprm* self, void* closure)
```

Comment:
Is this method made available for debugging purposes? 
Is there a use case for accessing this attribute from Python. 
I'm mostly curious how this is used.
There are other attributes in this category, like um `category`.

Reply:
> Is this method made available for debugging purposes?

Probably. I think this is probably one of the least useful methods. Added here for completeness of wrapping `prjprm`.

> Is there a use case for accessing this attribute from Python.

None I can think of.

> There are other attributes in this category, like um category.

I think `category` could, potentially, be quite useful. For the very least it is informative.